### PR TITLE
Feature 79/connect send registration request with backend

### PR DESCRIPTION
**Pull request by melinahernandeztheam**
## Description

Connect component AppReferralForm to RegisterRequestService to create a new RegisterRequest on form submit

## Changes

- Add RegisterRequestService to AppReferralForm
- Add recipientUserId to UserProps

## Checks

- [ ] Project Builds
- [ ] Project passes tests and checks
- [ ] Updated documentation accordingly

## Additional information

I had to merge this PR https://github.com/TheTributeCommunity/fesbal-frontend/pull/72 in this branch to get the RegistrationRequestService

### CODE REVIEW

**Comment by tainguyenbui** [comment]
From what I can see in several PRs, the file naming for services is not following convention. Should it be camel case or or kebab case?

File: src/mocks/users.mock.ts [hunk 1]
```diff
@@ -1,7 +1,9 @@
 import UserProps from "../types/UserProps";
+import { v4 as uuidv4 } from 'uuid';
 
 const users: UserProps[] = [
     {
+        recipientUserId: uuidv4(),
```
Comment (BorjaZarco) [inline src/mocks/users.mock.ts:6]: Why is there a recipientUserId and a id on a User?

Comment (melinahernandeztheam) [reply src/mocks/users.mock.ts:6]: because someone called id, to the DNI field

File: src/hooks/useUploadReferral.ts [hunk 2]
```diff
@@ -29,8 +32,13 @@ const useUploadReferral = () => {
     };
 
     const handleOnClick = (href: string) => {
-        console.log(href);
-        navigate(href);
+        if(usersMock[0].recipientUserId) {
```
Comment (BorjaZarco) [inline src/hooks/useUploadReferral.ts:35]: I believe this is done to be quick but getting the user this way? I mean, where would you get their data from if there was actually a user? 

Would it be stored in a redux store or do you get it from a service, would it be passed through a prop?

File: src/types/UserProps.ts [hunk 3]
```diff
@@ -1,4 +1,5 @@
 interface UserProps {
+    recipientUserId?: string;
```
Comment (BorjaZarco) [inline src/types/UserProps.ts:2]: Why is there a recipientUserId and an `id` on a User? Isn't the recipient a user with the role `Recipient`?

**Review by BorjaZarco — APPROVED** [review]

**Pull request merged**